src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Legacy/LegacyHostImplementation.cs

```diff
@@ -36,7 +36,7 @@ public static MarshalType GetMarshalTypeFromType(Type type)
                     case TypeCode.UInt64:
                         return MarshalType.ENUM64;
                     default:
-                        throw new JSException($"Unsupported enum underlying type {typeCode}");
+                        throw new ArgumentException(SR.Format(SR.UnsupportedEnumType, type.FullName));
```

Comment:
Can we add name of the argument for `ArgumentException`?

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception.-ctor?view=net-7.0#system-argumentexception-ctor(system-string-system-string)